### PR TITLE
Fix two problems with `unset -f` behavior

### PR DESCRIPTION
**Pull request by JohnoKing**
This pull request backports two bug fixes from ksh93v- for the problems detailed in #21. These fixes do the following:

In `src/cmd/ksh93/sh/name.c`, the check for if a function is still running has been corrected to fix a segmentation fault that occurred when a POSIX function unset itself.

The missing functionality for when a KornShell style function unsets itself has been added to `src/cmd/ksh93/sh/xec.c`. This kind of function will now be properly unset after it has finished running, rather than silently cause `unset -f` to fail.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -3,6 +3,14 @@ For full details, see the git log at: https://github.com/ksh93/ksh
 
 Any uppercase BUG_* names are modernish shell bug IDs.
 
+2020-06-17:
```
Comment (McDutchie) [inline NEWS:6]: Please add a blank line for formatting consistency.

File: src/cmd/ksh93/tests/functions.sh [hunk 2]
```diff
@@ -1249,5 +1249,10 @@ expect_status=0
 [[ $actual == "$expect" ]] ||
     err_exit "autoload function skipped dir test wrong output (expected $(printf %q "$expect"), got $(printf %q "$actual"))"
 
+# ======
+# When a function unsets itself, it should not fail to be unset
+$SHELL -c 'fn() { unset -f fn; true; }; fn' 2> /dev/null || err_exit 'unset of POSIX function in the calling stack fails'
+$SHELL -c 'function ftest { ftest2; }; function ftest2 { unset -f ftest; }; ftest' 2> /dev/null || err_exit 'unset of ksh function in the calling stack fails'
```
Comment (McDutchie) [inline src/cmd/ksh93/tests/functions.sh]: These tests are not complete. They check that the crash is gone, but they don't actually check whether unsetting the function succeeded.

There are various ways of checking that, but the most reliable is probably the brute approach: set `PATH=/dev/null`, then try to run the function a second time with `2>/dev/null` and check that the exit status is 127 (not found).

I'd also lose the global `2>/dev/null` on these tests. If `unset -f` ever prints an error message, I would want to know about that.

**Review by McDutchie — CHANGES_REQUESTED** [review]
I was half expecting you to show up for this one. :)  Couple of comments below…

File: src/cmd/ksh93/tests/functions.sh (hunk 2 above)
Comment (JohnoKing) [reply src/cmd/ksh93/tests/functions.sh]: I have made the requested changes to the two regression tests. I have also added another test as well for another potential bug in POSIX functions (although the new test does use `2> /dev/null` as it is expected to return an error).

File: NEWS (hunk 1 above)
Comment (JohnoKing) [reply NEWS:6]: This has been fixed.

File: src/cmd/ksh93/tests/functions.sh (hunk 2 above)
Comment (JohnoKing) [reply src/cmd/ksh93/tests/functions.sh]: Oops, I forgot to check the return status. One moment.

Comment (JohnoKing) [reply src/cmd/ksh93/tests/functions.sh]: The regression tests in this pull request (there are now four added) have been fully updated.
